data-reconciliation-app/docs/adr/01-data-ingest.md

```diff
@@ -2,126 +2,150 @@
 
 ## Decision
 
-## Model for K-V store
+## API
 
-```
-//psuedo code for model based off forum app
+### Endpoint
 
-// this is unique for each user
-type User = string;
+- Description
+  - Ingest member or user data into our application and add data to K-V store
+- Path
+  - /ingest
+- HTTP Method
+  - POST
+- URL Params
+  - N/A
+- Headers
+  - content-type: application/json
+- Request will be a `ccfapp.Request Object`. We will interrogate the `ccfapp.Request Object` to extract the user or member and [authenticate them via certficates](#security). We will read the request body as a JSON.
+- API Status Codes
+  - OK
+    - Status: 200
+    - Status Text: "Votes have been successfully recorded"
+  - UNAUTHORIZED
+    - Status: 401
+    - Status Text: "Unauthorized"
+  - BAD_REQUEST
+    - Status: 400
+    - Status Text: "Validation Error"
 
-interface AttributeBase {
-  votes: Record<User, T>;
-}
+### Security
 
-interface StringAttribute extends AttributeBase<string> {
-  type: "string";
-}
+Users will be authenticated via certificates, which is natively supported by the CCF framework.
+
+//TODO Add reference to this future ADR: https://github.com/microsoft/ccf-app-samples/issues/102
+
+## Models
 
-interface NumericAttribute extends AttributeBase<number> {
-  type: "number";
+### Data Schema Model
+
+The ingest model has been documented by [this ADR](./02-data-schema-strategy.md#option3-defining-data-schema-via-deployment). Our ingest data model will look like:
+
+```
+export interface DataSchema {
+    key: string;
+    value: string | number;
 }
+```
+
+### Data Record Model
 
-type Attribute = StringAttribute | NumericAttribute;
+Upon ingest, `DataSchema` records will be mapped to a `DataRecord` model. `DataRecord`s can be string or numeric.
 
-type AttributeMap = ccfapp.TypedKvMap<string, Attribute>;
 ```
+export type DataAttributeType = string | number;
+export interface DataRecordProps {
+  key: string;
+  value: DataAttributeType;
+}
+```
+
+### Repository Model
 
-### Sample Ingest Data
+Our repository will be a K-V store. Our K-V store will be defined as:
 
-For example, from Member 1:
+```
+const kvStore = ccfapp.typedKv(
+  "data",
+  ccfapp.string,
+  ccfapp.json<ReconciledRecord>()
+);
+```
 
-| id    | company_name |
-| ----- | ------------ |
-| A0128 | microsoft    |
-| A0129 | google       |
-| A0130 | amazon       |
+A `ReconciledRecord` represents all of users who submittted data on the record and their opinion of the record.
 
-For example, from Member 2:
-| id | company_name |
-|---|---|
-| A0128 | microsoft |
-|A0129 | alphabet |
-| A0130 | amazon |
+```
+export class ReconciledRecord implements ReconciledRecordProps {
+  key: string;
+  values: ReconciliationMap = {};
+```
 
-### Domain Model Sample - String Attribute
+The `ReconciliationMap` is a map where key is the user ID and value is the record submitted by the user. For example,
```

Comment:
Is `A0129` a User ID or Record ID (Institution ID even?) 😅 ?

Reply:
A0129 is a record ID in this example. User ID is "Member 1" here.